### ISSUE TITLE
A little confuse about the code.

### ISSUE THREAD

**Issue by lililuya** [Junhua-Liao/Light-ASD#25]
Hi, thanks for your great work. I am little confuse about the inference implement code. The code part are as follow
```python
def evaluate_network(files, args):
	# GPU: active speaker detection by pretrained model
	s 			= ASD()
	s.loadParameters(args.pretrainModel)
	sys.stderr.write("Model %s loaded from previous state! \r\n"%args.pretrainModel)
	s.eval()
	allScores   = []
	# durationSet = {1,2,4,6} # To make the result more reliable
	durationSet = {1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6} # Use this line can get more reliable result
	# 所有的trackers
	for file in tqdm.tqdm(files, total = len(files)):
		fileName 	 = os.path.splitext(file.split('/')[-1])[0] # Load audio and video
		_, audio	 = wavfile.read(os.path.join(args.pycropPath, fileName + '.wav'))
		# 整个音频的特征
		# (NUMFRAMES, numcep)
		# 16KHZ-->0.025*16000=400(samples)
		# NUMFRAMES=[audio_length - winlen] / win_step
		# 1s aduio--> 16000 samples-->NUMFRAMES = (1 - 0.025)/0.01 + 1=98 
		# audio_length = 1s= 25fps =（NumFrames-1)/100 + 0.025
		audioFeature = python_speech_features.mfcc(audio, 16000, numcep = 13, winlen = 0.025, winstep = 0.010) 
		video 		 = cv2.VideoCapture(os.path.join(args.pycropPath, fileName + '.avi'))
		videoFeature = []
		while video.isOpened():
			ret, frames = video.read()
			if ret == True:
				face = cv2.cvtColor(frames, cv2.COLOR_BGR2GRAY)
				face = cv2.resize(face, (224,224))
				face = face[int(112- (112/2)) : int(112 + (112/2)), int(112 - (112/2)) : int(112 + (112/2))]
				videoFeature.append(face)
			else:
				break
		video.release()
		videoFeature = numpy.array(videoFeature) 
		length 		 = min((audioFeature.shape[0] - audioFeature.shape[0] % 4) / 100, videoFeature.shape[0])
		audioFeature = audioFeature[:int(round(length * 100)), :]
		videoFeature = videoFeature[:int(round(length * 25)),: ,:]
		allScore = [] 
		for duration in durationSet:
			batchSize = int(math.ceil(length / duration))
			print(batchSize)
			scores = []
			with torch.no_grad():
				for i in range(batchSize):
					inputA = torch.FloatTensor(audioFeature[i * duration * 100: (i+1) * duration * 100,:]).unsqueeze(0).cuda()
					inputV = torch.FloatTensor(videoFeature[i * duration * 25: (i+1) * duration * 25,:,:]).unsqueeze(0).cuda()
					embedA = s.model.forward_audio_frontend(inputA)
					embedV = s.model.forward_visual_frontend(inputV)	
					out    = s.model.forward_audio_visual_backend(embedA, embedV)
					score  = s.lossAV.forward(out, labels = None)
					scores.extend(score)
			allScore.append(scores)
		allScore = numpy.round((numpy.mean(numpy.array(allScore), axis = 0)), 1).astype(float)
		allScores.append(allScore)	
	return allScores
```
1.The length choice between the `audioFeature` and `videoFeature `, The python_speech_features.mfcc return the feature vector of shape `(NUMFRAMES, numcep)`  refer from the [source](https://python-speech-features.readthedocs.io/en/latest/). The videoFeature's first dim is representing the num of cropped video clip frames. For example, 1s to 25fps, while the audioFeature's first dim can be calculated by $$\text{NUMFRAMES} = \left\lfloor \frac{\text{signal_length} - \text{frame_length}}{\text{frame_step}} \right\rfloor + 1 $$
.So, the same as 1s audio to 98 audio frames. Why we need calculate this `(audioFeature.shape[0] - audioFeature.shape[0] % 4) / 100` ?
2.The explaination of duationSet, why `Use this line can get more reliable result` ?
Any help you provide will be greatly appreciated!  Looking forward your response!

**Comment by lililuya** [comment]
Hi, I read the code carefully. I guess that the code line `min((audioFeature.shape[0] - audioFeature.shape[0] % 4) / 100, videoFeature.shape[0])` is intented to compare the audio and video feature time length. The `videoFeature.shape[0]` maybe `videoFeature.shape[0]/25`. 
Besides, the relationship between the audio feature and timeline I think it's 98(feature vector )/s. For example, 1s audio, `(1-0.025)/0.01=98`. I wonder if I missing something related to the result. Looking forward for  your response!

**Comment by ztkevlee** [comment]
> Hi, I read the code carefully. I guess that the code line `min((audioFeature.shape[0] - audioFeature.shape[0] % 4) / 100, videoFeature.shape[0])` is intented to compare the audio and video feature time length. The `videoFeature.shape[0]` maybe `videoFeature.shape[0]/25`. Besides, the relationship between the audio feature and timeline I think it's 98(feature vector )/s. For example, 1s audio, `(1-0.025)/0.01=98`. I wonder if I missing something related to the result. Looking forward for your response!

I agree `videoFeature.shape[0]` should be `videoFeature.shape[0]/25` instead, so as to align with the `TalkNet-ASD` implementation:
https://github.com/TaoRuijie/TalkNet-ASD/blob/main/demoTalkNet.py#L233

**Comment by kev-leehc** [comment]
> Hi, I read the code carefully. I guess that the code line `min((audioFeature.shape[0] - audioFeature.shape[0] % 4) / 100, videoFeature.shape[0])` is intented to compare the audio and video feature time length. The `videoFeature.shape[0]` maybe `videoFeature.shape[0]/25`. Besides, the relationship between the audio feature and timeline I think it's 98(feature vector )/s. For example, 1s audio, `(1-0.025)/0.01=98`. I wonder if I missing something related to the result. Looking forward for your response!

I agree `videoFeature.shape[0]` should be `videoFeature.shape[0]/25` instead, so as to align with the `TalkNet-ASD` implementation:
https://github.com/TaoRuijie/TalkNet-ASD/blob/main/demoTalkNet.py#L233
